contracts/identity/lib.rs

```diff
@@ -354,6 +386,44 @@ mod identity {
 			Ok(())
 		}
 
+		#[ink(message)]
+		pub fn set_recovery_account(&mut self, recovery_account: AccountId) -> Result<(), Error> {
+			let caller = self.env().caller();
+			ensure!(self.identity_of.get(caller).is_some(), Error::NotAllowed);
+
+			let identity_no = self.identity_of.get(caller).unwrap();
+
+			self.recovery_account_of.insert(identity_no, &recovery_account);
+			self.env().emit_event(RecoveryAccountSet { identity_no, recovery_account });
+
+			Ok(())
+		}
+
+		/// Transfers the ownership of an identity to another account.
+		///
+		/// Only callable by the identity owner or any account that the identity
+		/// owner added as a proxy.
+		#[ink(message)]
+		pub fn transfer_ownership(
+			&mut self,
+			identity_no: IdentityNo,
+			new_owner: AccountId,
+		) -> Result<(), Error> {
+			let caller = self.env().caller();
+
+			let is_recovery_account = self.recovery_account_of.get(identity_no) == Some(caller);
+			let Some(identity_owner) = self.owner_of(identity_no) else { return Err(Error::NotAllowed) };
+
+			ensure!(identity_owner == caller || is_recovery_account, Error::NotAllowed);
+
+			self.identity_of.remove(identity_owner);
+			self.identity_of.insert(new_owner, &identity_no);
+
+			self.owner_of.insert(identity_no, &new_owner);
+
```

Comment:
`new_owner` should not have an `Identity` created.

Reply:
I am not sure I understand what you mean by this.

Reply:
```rust
self.identity_of.insert(new_owner, &identity_no);
```

If `new_owner` hasn't created an identity yet, it will be okay.
What if `new_owner` already has an identity?